### PR TITLE
Fixes #9232: Add an API to list shared files properties

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/9232

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+      if (Files.isRegularFile(path)) {
+        val data = Files.readAllBytes(path)
+        val headers =
+         ("Content-type" -> "application/octet-stream") ::
+         ("Content-length" -> data.length.toString) ::
+         ("Content-disposition" -> s"attachment; filename=${path.getFileName}") ::
+         Nil
+
+        Full(StreamingResponse(new ByteArrayInputStream(data),() => {}, data.length, headers, Nil, 200))
+      } else {
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala]: Here, you are reading the file into memory to then stream it to the network. You should side-step the memory part to avoid to consume huge quantity of RAM on ISO.
It's easy to get the file size without reading it with (new File(path)).length (perhaps there is an quivalent method on nio). 
And for the streaming part, a simple FileIntputStream should be ok:
val file = new File(path)
Full(StreamingResponse(new FileInputStream(file),() => {}, file.length, headers, Nil, 200))

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+          response
+        case eb : EmptyBox =>
+          val fail = eb ?~! s"An error occured while looking into directory"
+          logger.error(fail.messageChain)
+          errorResponse(fail.messageChain)
+      }
+
+    }
+  }
+  serve("secure" / "api" / s"sharedfile" prefix requestDispatch)
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala]: why the s"" for sharedFile ?

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        Failure(s"File '${path}' is not a directory")
+      }
+    } else {
+      Failure(s"File '${path}' does not exist")
+    }
+  }
+
+  def fileContent(path : Path) : Box[LiftResponse] = {
+    if (Files.exists(path)) {
+      if (Files.isRegularFile(path)) {
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala]: I think we should check if the file is binary or text.

Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/rest/sharedFiles/SharedFilesAPI.scala]: That will be done in another PR

**Comment by VinceMacBuche** [comment]
PR rebased
